server/sonar-server/src/test/java/org/sonar/server/edition/StandaloneEditionManagementStateImplTest.java

```diff
@@ -277,6 +278,91 @@ public void startAutomaticInstall_fails_with_ISE_if_called_after_manualInstall()
   }
 
   @Test
+  public void uninstall_fails_with_ISE_if_not_started() {
+    expectISENotStarted();
+
+    underTest.uninstall();
+  }
+
+  @Test
+  public void uninstall_sets_resets_fields_after_start() {
+    underTest.start();
+
+    PendingStatus newStatus = underTest.uninstall();
+
+    assertThat(newStatus).isEqualTo(UNINSTALL_IN_PROGRESS);
+    assertThat(underTest.getPendingInstallationStatus()).isEqualTo(UNINSTALL_IN_PROGRESS);
+    assertThat(underTest.getPendingEditionKey()).isEmpty();
+    assertThat(underTest.getPendingLicense()).isEmpty();
+    assertThat(underTest.getCurrentEditionKey()).isEmpty();
```

Comment:
depending on whether you rebase on my change or the other way around, a test on the error message should be added here